### PR TITLE
feat/task

### PR DESCRIPTION
**Pull request by shockch4rge**
- `Todo` model still requires a `board` relation in a future commit.

@TechSupportz

### CODE REVIEW

**Comment by TechSupportz** [comment]
Just a few thoughts
- I think we should rename `todo` to `task` so that it is easier to understand.
- `Todo` seems to be missing a primary key
- The User entity is also missing a `@OneToMany` decorator which I think is needed

@Shockch4rge could you make these changes while I work on boards? :D

**Comment by shockch4rge** [comment]
Issues fixed.

File: src/todo/TaskController.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Body, Controller, Delete, Get, Path, Post, Put, Route } from "tsoa";
+
+import { TaskCreate, TaskDelete, TaskGet } from "./Task";
+import { TodoService as TaskService } from "./TaskService";
+
+import type { TaskUpdate } from "./Task";
+
+@Route("todos")
```
Comment (TechSupportz) [inline src/todo/TaskController.ts]: Route should be tasks

File: src/todo/TaskController.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        this.taskService = new TaskService();
+    }
+
+    @Get(`{id}`)
+    public async get(@Path() id: TaskGet) {
+        return this.taskService.get(id);
+    }
+
+    @Post()
+    public async create(@Body() todo: TaskCreate) {
```
Comment (TechSupportz) [inline src/todo/TaskController.ts]: Params here should be task as well

File: src/todo/TaskController.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        return this.taskService.create(todo);
+    }
+
+    @Put(`{id}`)
+    public async update(@Path() id: TaskUpdate["id"], @Body() todo: Omit<TaskUpdate, "id">) {
```
Comment (TechSupportz) [inline src/todo/TaskController.ts]: Params here should be task here too

File: src/todo/TaskService.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { uuid } from "../util/uuid";
+import { Task } from "./Task";
+
+import type { TaskCreate, TaskDelete, TaskUpdate } from "./Task";
+
+export class TodoService {
```
Comment (TechSupportz) [inline src/todo/TaskService.ts]: This should be `TaskService`

File: src/todo/TaskService.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    async get(id: Task["id"]) {
+        return Task.findOneBy({ id });
+    }
+
+    async create(todo: TaskCreate) {
```
Comment (TechSupportz) [inline src/todo/TaskService.ts]: params should be task

File: src/todo/TaskService.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        return Task.create({
+            ...todo,
+            id: uuid(),
+            status: "open",
+        });
+    }
+
+    async update(todo: TaskUpdate) {
```
Comment (TechSupportz) [inline src/todo/TaskService.ts]: params should be task

File: src/todo/TaskController.ts (hunk 1 above)
Comment (Shockch4rge) [reply src/todo/TaskController.ts]: fixed

**Review by TechSupportz — APPROVED** [review]
Everything looks good

**Pull request merged**